### PR TITLE
Docs: updated Creating a Java API for a Web Component

### PR DESCRIPTION
**Pull request by MarcinVaadin**
Updated 'Creating a Java API for a Web Component' with LitComponent instead of Polymer, updated examples referring to current addon-template, updated external links.

### CODE REVIEW

**Comment by jouni** [comment]
@MarcinVaadin, @mshabarov, can you get someone else than Teppo to review this?

File: articles/create-ui/web-components/index.asciidoc [hunk 1]
```diff
@@ -5,145 +5,105 @@ order: 110
...
 
-* to load the HTML/JS/CSS files needed by the component (see the instructions below);
-* a Java API to configure the component and listen to events from it.
-See <<java-api-for-a-web-component#,Creating a Java API for a Web Component>> for more information.
+The Web Component's client-side files -- typically JavaScript module files -- are available using https://www.npmjs.com/[npm], which Vaadin supports. It automatically installs and uses `npm` packages. It also serves the static files to the browser.
 
-The Web Component's client-side files (typically JavaScript module files) are available using https://www.npmjs.com/[npm].
-Vaadin supports `npm` and automatically installs and uses `npm` packages.
-It also serves the static files to the browser.
+The `pnpm` tool is used for package management instead of `npm`, which happens transparently. However, if there is a reason to use `npm` instead of `pnpm`, you may disable `pnpm` (see <<{articles}/configuration/properties#,Configuration Properties>> or <<{articles}/production/production-build#plugin-goals-and-goal-parameters,Production Mode Plugin Goals and Parameters>>).
```
Comment (mcollovati) [inline articles/create-ui/web-components/index.asciidoc:14]: Is this still true? I think the default has been switched back to npm

**Review by mcollovati — COMMENTED** [review]
Added a question about pnpm usage.
Other than that, LGTM

File: articles/create-ui/web-components/index.asciidoc (hunk 1 above)
Comment (tepi) [reply articles/create-ui/web-components/index.asciidoc:14]: npm is the default

**Review by Mikaelsu — APPROVED** [review]

**Pull request merged**